src/node/node.js

```diff
@@ -713,6 +713,27 @@ class Node extends Base {
         return peer;
     }
 
+    /**
+     * Ping a peer
+     * @param {Peer} peer
+     * @returns {Promise<Peer, number>}
+     */
+    pingPeer (peer) {
+        return new Promise(resolve => {
+            const hrTimeStart = (hrTime => hrTime[0] * 1000000 + hrTime[1] / 1000)(process.hrtime());
```

Comment:
You can simplify the code to calculate `pingInMs` to 

```
const hrstart = process.hrtime()

setTimeout(function (argument) {
    // execution time simulated with setTimeout function
    const hrend = process.hrtime(hrstart)

    console.info("Execution time (hr): %ds %dms", hrend[0], hrend[1]/1000000);
}, 1);
```

Reply:
Thanks a lot, I did not know hrtime could return a difference! Will definitely take a look.